File: app/src/main/java/it/paranoidsquirrels/idleguildmaster/storage/data/quests/instances/ClashOfTitans.java
package it.paranoidsquirrels.idleguildmaster.storage.data.quests.instances;

import it.paranoidsquirrels.idleguildmaster.R;
import it.paranoidsquirrels.idleguildmaster.storage.data.quests.Quest;
import it.paranoidsquirrels.idleguildmaster.storage.data.quests.QuestsManager;

/* JADX INFO: loaded from: classes3.dex */
public class ClashOfTitans extends Quest {
    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.quests.Quest
    public void calculateTargetProgress(int i) {
        this.targetProgress = (((long) (i - 9)) * 25) + 50;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.quests.Quest
    protected void configure() {
        this.idName = R.string.quest_clash_of_titans_name;
        this.idDescription = R.string.quest_clash_of_titans_description;
        this.defaultRarity = 2;
        this.minimumDifficulty = 9;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.quests.Quest
    public void realignStaticReference() {
        QuestsManager.clashOfTitans = this;
    }
}

File: app/src/main/kotlin/it/paranoidsquirrels/idleguildmaster/storage/data/quests/instances/ClashOfTitans.kt
package it.paranoidsquirrels.idleguildmaster.storage.data.quests.instances

import it.paranoidsquirrels.idleguildmaster.R
import it.paranoidsquirrels.idleguildmaster.storage.data.quests.Quest
import it.paranoidsquirrels.idleguildmaster.storage.data.quests.QuestsManager

class ClashOfTitans : Quest() {
    override fun calculateTargetProgress(difficulty: Int) {
        targetProgress = (((difficulty - 9).toLong() * 25) + 50).toLong()
    }

    override fun configure() {
        idName = R.string.quest_clash_of_titans_name
        idDescription = R.string.quest_clash_of_titans_description
        defaultRarity = 2
        minimumDifficulty = 9
    }

    override fun realignStaticReference() {
        QuestsManager.clashOfTitans = this
    }
}
